### ISSUE TITLE
Force insert mode in all magit buffers

### ISSUE THREAD

**Issue by pierwill** [xahlee/xah-fly-keys#49]
Hi! Love xah-fly-keys!

Using with magit is very tough. One has to activate insert mode in every new popup buffer. One way to fix would be to force insert mode for all magit buffers...

**Comment by xahlee** [comment]
that issue is hard to fix. Mostly because most magit keys are text properties. that is, the key depends on where your cursor is.
There's no clean way to override that. Even so, it's a problem of how one'd recover magit's keys.

as a workaround by automatically enable insert mode... am not 100% sure about that, since magit isn't bundled with emacs. It's easy to have personal init to do that by a hook.

**Comment by divansantana** [comment]
doing this helps for me.

```lisp
  (global-set-key (kbd "<escape>") 'xah-fly-mode-toggle)
```
So `escape` for me toggles it on and off. And works in magit so when it gets in a state I use this to fix it.

**Comment by pierwill** [comment]
I use ESC for `keyboard-quit`, but yes that would be a good partial solution. The thing is, magit uses lots of pop-up buffers; do you need to hit ESC for each new popup buffer as well?

**Comment by xahlee** [comment]
@pierwill can you show perhaps screenshots of some of the magit buffers?

i use magit few times a week, but only the very basic like magit status, commit, and sometimes log.

maybe try this `(add-hook 'magit-mode-hook 'xah-fly-insert-mode-activate)` and see how it goes.

(i was thinking making it work for other magit buffers... but not sure what they are offhand. maybe the magit-mode-hook will work for all of them, but maybe not, so will need to know what buffers/mode it uses.)

**Comment by divansantana** [comment]
@pierwill I have this in the `use-package` declaration

```
  :hook	(magit-status-mode . xah-fly-insert-mode-activate)
```
Though in some cases it still goes into command mode.

I do that for many modes. It's not ideal.

**Comment by gcentauri** [comment]
I had an issue with some popups still switching to command mode.  `fetch` is the only one coming to mind right now.  it helped to add:

`  (add-hook 'magit-popup-mode-hook 'xah-fly-insert-mode-activate)`

**Comment by xahlee** [comment]
fixed by @DanLanglois  and  @wi11dey  commit #89
